flat/custom/bsc/LinkTokenBEP20.sol

```diff
@@ -0,0 +1,1614 @@
+   *
+   * This is an alternative to {approve} that can be used as a mitigation for
+   * problems described in {IERC20-approve}.
+   *
+   * Emits an {Approval} event indicating the updated allowance.
+   *
+   * Requirements:
+   *
+   * - `spender` cannot be the zero address.
+   */
+  function increaseApproval(address spender, uint256 addedValue) public virtual returns (bool) {
+    return super.increaseAllowance(spender, addedValue);
+  }
+
+  /**
+   * @dev Atomically decreases the allowance granted to `spender` by the caller.
+   *
+   * This is an alternative to {approve} that can be used as a mitigation for
+   * problems described in {IERC20-approve}.
+   *
+   * Emits an {Approval} event indicating the updated allowance.
+   *
+   * Requirements:
+   *
+   * - `spender` cannot be the zero address.
+   * - `spender` must have allowance for the caller of at least
+   * `subtractedValue`.
+   */
+  function decreaseApproval(address spender, uint256 subtractedValue) public virtual returns (bool) {
+    return super.decreaseAllowance(spender, subtractedValue);
+  }
+}
+
+// File: link_token/contracts/v0.6/token/ERC677.sol
+
+pragma solidity ^0.6.0;
+
+
+abstract contract ERC677 is IERC20 {
+  function transferAndCall(address to, uint value, bytes memory data) public virtual returns (bool success);
+
+  event Transfer(address indexed from, address indexed to, uint value, bytes data);
+}
+
+// File: link_token/contracts/v0.6/token/ERC677Receiver.sol
+
+pragma solidity ^0.6.0;
+
+abstract contract ERC677Receiver {
+  function onTokenTransfer(address _sender, uint _value, bytes memory _data) public virtual;
+}
+
+// File: link_token/contracts/v0.6/ERC677Token.sol
+
+pragma solidity ^0.6.0;
+
+
+
+
+abstract contract ERC677Token is ERC20, ERC677 {
+  /**
+   * @dev transfer token to a contract address with additional data if the recipient is a contact.
+   * @param _to The address to transfer to.
+   * @param _value The amount to be transferred.
+   * @param _data The extra data to be passed to the receiving contract.
+   */
+  function transferAndCall(address _to, uint _value, bytes memory _data)
+    public
+    override
+    virtual
+    returns (bool success)
+  {
+    super.transfer(_to, _value);
+    emit Transfer(msg.sender, _to, _value, _data);
+    if (isContract(_to)) {
+      contractFallback(_to, _value, _data);
+    }
+    return true;
+  }
+
+
+  // PRIVATE
+
+  function contractFallback(address _to, uint _value, bytes memory _data)
+    private
+  {
+    ERC677Receiver receiver = ERC677Receiver(_to);
+    receiver.onTokenTransfer(msg.sender, _value, _data);
+  }
+
+  function isContract(address _addr)
+    private
+    view
+    returns (bool hasCode)
+  {
+    uint length;
+    assembly { length := extcodesize(_addr) }
+    return length > 0;
+  }
+}
+
+// File: link_token/contracts/v0.6/LinkToken.sol
+
+pragma solidity ^0.6.0;
+
+
+
+contract LinkToken is LinkERC20, ERC677Token {
+  uint private constant TOTAL_SUPPLY = 10**27;
```

Comment:
We would like to set it to 1000000*1e18 for now.

Reply:
This is the artifact of the Ethereum Mainnet LinkToken we extend. If you check the `_onCreate` function for `ChildLinkToken` you will see that the actual child token total supply starts with 0.

Let's discuss this on the call on Monday.